src/net_processing.cpp

```diff
@@ -3709,20 +3709,16 @@ void PeerManager::ProcessMessage(CNode& pfrom, const std::string& msg_type, CDat
 
         // Nodes must NEVER send a data item > 520 bytes (the max size for a script data object,
         // and thus, the maximum size any matched object can have) in a filteradd message
-        bool bad = false;
         if (vData.size() > MAX_SCRIPT_ELEMENT_SIZE) {
-            bad = true;
+            pfrom.fDisconnect = true;
         } else if (pfrom.m_tx_relay != nullptr) {
             LOCK(pfrom.m_tx_relay->cs_filter);
             if (pfrom.m_tx_relay->pfilter) {
                 pfrom.m_tx_relay->pfilter->insert(vData);
             } else {
-                bad = true;
+                pfrom.fDisconnect = true;
```

Comment:
This behavior seems to have previously triggered log messages "too-large bloom filter` or  `bad filteradd message`. I see that in other instances of `pfrom.fDisconect` it seems to be often preceded by log messages, e.g. `LogPrintf("Disconnecting outbound peer %d -- headers chain has insufficient work\n", pfrom.GetId());`.

Should there be a log message here for the disconnect as well?